File: AGTIA_Objective-C/AGTIACircleView/AGTIACircleView/CircleView/CircleView.m
//
//  CircleView.m
//  AGTIACircleView
//
//  Created by fernando on 2017/2/3.
//  Copyright © 2017年 fernando. All rights reserved.
//

#import "CircleView.h"

@implementation CircleView

+ (Class)layerClass {
    return [CircleLayer class];
}

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        self.circleLayer = [CircleLayer layer];
        self.circleLayer.frame = CGRectMake(0, 0, frame.size.width, frame.size.height);
        self.circleLayer.contentsScale = [UIScreen mainScreen].scale;
        [self.layer addSublayer:self.circleLayer];
    }
    return self;
}

@end

File: AGTIA_Swift/AnimatedCircle/AnimatedCircle/CircleView/CircleView.swift
//
//  CircleView.swift
//  AnimatedCircle
//
//  Created by fernando on 2017/2/3.
//  Copyright © 2017年 fernando. All rights reserved.
//

import UIKit

class CircleView: UIView {
    
    var circleLayer = CircleLayer()
    override init(frame: CGRect) {
        super.init(frame: frame)
        circleLayer.frame = CGRect(x: 0, y: 0, width: frame.size.width, height: frame.size.height)
        circleLayer.contentsScale = UIScreen.main.scale
        layer.addSublayer(circleLayer)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
